### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DmytroBondariev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+
+    def wrapper(*args) -> int:
+        if str(args) not in cache_dict.keys():
+            print("Calculating new result")
+            result = func(*args)
+            cache_dict[str(args)] = result
+            return cache_dict[str(args)]
+        else:
+            print("Getting from cache")
+            return cache_dict[str(args)]
```
Comment (Grayder0152) [inline app/main.py:15]:
```suggestion
            cache_dict[str(args)] = result
        else:
            print("Getting from cache")
        return cache_dict[str(args)]
```

**Review by Grayder0152 — APPROVED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> int:
+    cache_dict = {}
+
+    def wrapper(*args) -> int:
+        if str(args) not in cache_dict.keys():
```
Comment (ArsenPidhoretskyi) [inline app/main.py:8]: No need to check as str. Just use if args not in

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
+            print("Calculating new result")
+            result = func(*args)
+            cache_dict[str(args)] = result
```
Comment (ArsenPidhoretskyi) [inline app/main.py:11]: write inline this
and also do not use str(args), just args

Comment (ArsenPidhoretskyi) [reply app/main.py:11]: Because for example someone call with function with parameter=1 and will have another result if call with "1"

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Changes Requested

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**